[Avalanche Developer Contest] ERC-20 as Subnet Native Token

zicoz18 opened the pull request:
Avalanche Developer Contests 2022 "How to use a ERC-20 C-chain token as the gas fee token — (Native Minter Precompile) + Bridge to call this contract" round 2 submission. Love to have some reviews.

jpop32 commented:
HI there!

Thank you for your submission, and congratulations on making it to Round 2! In this round, the idea is to polish the tutorial and make certain changes to make it more accessible, more in line with our needs and more uniform in looks with the rest of the material we have on our docs pages.

Once other reviewers from our team have had a chance to look over the material in more detail they will post their suggestions as well, but for now here are some general guidelines to check for:
* large (more than a dozen or lines or so) source files should be linked to, not embedded. It's ok to highlight outtakes of the most important parts of the file, but just presenting the whole file so that it can be copied is best avoided
* alerts, notes and warnings should be done using Markdown tags as shown [here](https://github.com/ava-labs/avalanche-docs/blob/master/docs/nodes/maintain/node-backup-and-restore.md#nodeid).

Some specific notes for your tutorial:
* one of the tasks in the tutorial is to provide overview and suggestions regarding the security and maintenance of the bridge. Could you add some material to cover that aspect or running the bridge?
* included videos does not seem to render, can you please check what's the issue?

yulin-dong reviewed (changes requested):
Please follow these guidelines and make appropriate changes:
- All section titles should be title cases, try https://titlecase.com/ to convert/confirm.
- All word `Subnet` should have letter capitalized.

Also please make sure you actually build and run your project before committing you changes.

yulin-dong commented on docs/community/tutorials-contest/2022/erc20-as-subnet-gas-token/README.md:
Please use :::alert instead.

rajranjan0608 commented on docs/community/tutorials-contest/2022/erc20-as-subnet-gas-token/README.md:
👍

zicoz18 replied:
I guess this is an approval for `Security and Maintenance` content 🥳 🥳

rajranjan0608 replied:
Yup. Looks good to me. Readers will appreciate any reference links for further reading 😉

rajranjan0608 commented:
Hi @zicoz18. I have requested a few minor nits. Please have a look. Thank you.

zicoz18 commented on docs/community/tutorials-contest/2022/erc20-as-subnet-gas-token/README.md:
I used v18.6.0 when developing it. After your review, I have tried running it with v16.0.0 and v18.7.0 they seem fine.

rajranjan0608 replied:
Nice. Please add the version requirements for `npm` and `Hardhat` as well.

zicoz18 commented on docs/community/tutorials-contest/2022/erc20-as-subnet-gas-token/README.md:
I want to make sure I understand you correctly. You are suggesting I upload AvaxToken.sol rather than AvaxToken.sol.md and link AvaxToken.sol inside the tutorial right? If I understood you correctly, I have done the change you have suggested and ran it locally. It seems worse for me therefore, I wanted to make sure I am doing the right thing. I am sharing you a video of me running it locally and using both AvaxToken.sol and AvaxToken.sol.md. I am looking forward for your feedback

https://user-images.githubusercontent.com/65618011/181860125-1ff112dd-9f4c-4fba-bcf8-216796e878cf.mov

.

rajranjan0608 replied:
Hmm. It looks like it's not rendering well. You can keep the same version since I want to generate the same thing without any complexity of parsing code files to markdown—so no need to change anything here 🙂

zicoz18 commented:
After your review, I have looked and followed the tutorial again, but this time with fresh eyes. I have found some things that I would like to change. I have made those changes and now I am looking forward for your reply about your change requests.

zicoz18 commented:
Hi @rajranjan0608 I have made some comments but forgot to mention you 😔. Sorry if timing is not good but wanted to make sure I have mentioned you.

rajranjan0608 commented:
> Hi @rajranjan0608 I have made some comments but forgot to mention you 😔. Sorry if timing is not good but wanted to make sure I have mentioned you.

No worries 🙂. And sorry for the delay 🙏

zicoz18 commented:
Hey @rajranjan0608 I have adjusted according to your feedbacks. Added references part for further readings. I hope I have referenced them properly. Thanks for your feedbacks
